src/mongoc/mongoc-uri.c

```diff
@@ -737,13 +738,13 @@ _mongoc_uri_build_write_concern (mongoc_uri_t *uri) /* IN */
          value = bson_iter_int32 (&iter);
 
          switch (value) {
-         case -1:
-            mongoc_write_concern_set_w (write_concern,
-                                        MONGOC_WRITE_CONCERN_W_ERRORS_IGNORED);
-            break;
-         case 0:
-            mongoc_write_concern_set_w (write_concern,
-                                        MONGOC_WRITE_CONCERN_W_UNACKNOWLEDGED);
+         case MONGOC_WRITE_CONCERN_W_ERRORS_IGNORED:
+         case MONGOC_WRITE_CONCERN_W_UNACKNOWLEDGED:
+            /* Warn on conflict, since write concern will be validated later */
+            if (mongoc_write_concern_get_journal(write_concern)) {
+               MONGOC_WARNING("Journal conflicts with w value [w=%d].", value);
```

Comment:
When running tests, this emits:

```
WARNING:       mongoc: Journal conflicts with w value [w=0].
WARNING:       mongoc: Journal conflicts with w value [w=-1].
```

Since `mongoc_uri_new()` really doesn't give us an error reporting mechanism, I think warnings are helpful. Is there a convenient way to suppress the log output during the test, though? I saw some other tests using `mongoc_log_set_handler()`, but that was through a `main()` function that the URI test doesn't have, so I'm not sure of the best way to go about that.